frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -175,10 +175,19 @@ export const stopPipelineRun: KebabAction = (kind: K8sKind, pipelineRun: Pipelin
   return {
     label: 'Stop',
     callback: () => {
-      k8sUpdate(PipelineRunModel, {
-        ...pipelineRun,
-        spec: { ...pipelineRun.spec, status: 'PipelineRunCancelled' },
-      });
+      k8sPatch(
+        PipelineRunModel,
+        {
+          metadata: { name: pipelineRun.metadata.name, namespace: pipelineRun.metadata.namespace },
+        },
+        [
+          {
+            op: 'replace',
+            path: `/spec`,
+            value: { ...pipelineRun.spec, status: 'PipelineRunCancelled' },
+          },
```

Comment:
A patch request should be as small as possible only patching the parts of the resource you are changing.
```
          {
            op: 'replace',
            path: `/spec/status`,
            value: 'PipelineRunCancelled',
          },
```

Reply:
updated, the only problem that could occur though is, If someone starts a PipelineRun and stops it immediately, before the status field is generated, the patch might not work, but it is a very edge case.